code/datums/status_effects/xeno_buffs.dm

```diff
@@ -76,19 +76,20 @@
 	RegisterSignal(link_target, COMSIG_MOVABLE_MOVED, .proc/handle_dist)
 	RegisterSignal(link_owner, COMSIG_XENOMORPH_RESIN_JELLY_APPLIED, .proc/share_jelly)
 	RegisterSignal(link_target, COMSIG_XENOMORPH_RESIN_JELLY_APPLIED, .proc/share_jelly)
-	RegisterSignal(link_owner, COMSIG_XENOMORPH_HEALED_BY_ABILITY, .proc/share_heal)
-	RegisterSignal(link_target, COMSIG_XENOMORPH_HEALED_BY_ABILITY, .proc/share_heal)
+	RegisterSignal(link_owner, COMSIG_XENOMORPH_BRUTE_DAMAGE, .proc/share_heal)
+	RegisterSignal(link_target, COMSIG_XENOMORPH_BRUTE_DAMAGE, .proc/share_heal)
+	RegisterSignal(link_owner, COMSIG_XENOMORPH_BURN_DAMAGE, .proc/share_heal)
+	RegisterSignal(link_target, COMSIG_XENOMORPH_BURN_DAMAGE, .proc/share_heal)
```

Comment:
You can register to mutiple signals at once

With a regidtersignal(source, list(sig1, sig2), proc)

Reply:
Donezo.